src/main.rs

```diff
@@ -32,6 +33,64 @@ type PackageName<'a> = &'a str;
 type RootFeature<'a> = (PackageName<'a>, Feature<'a>);
 
 fn main() {
+    let args: Vec<String> = std::env::args().collect();
+    let args: Vec<&str> = args.iter().map(AsRef::as_ref).collect();
+
+    match args.as_slice() {
+        [_, "--stdout"] => generate_cargo_nix(io::stdout()),
+        [_, "-s"] => generate_cargo_nix(io::stdout()),
+        [_, "--file"] => write_to_file("Cargo.nix"),
+        [_, "-f"] => write_to_file("Cargo.nix"),
+        [_, "--file", file] => write_to_file(file),
+        [_, "-f", file] => write_to_file(file),
+        [_] => print_help(),
+        [_, "--help"] => print_help(),
+        [_, "-h"] => print_help(),
+        [_, "--version"] => print_version(),
+        [_, "-v"] => print_version(),
+        _ => {
+            println!("Invalid arguments: {:?}", &args[1..]);
+            println!("\nTry again, with help: \n");
+            print_help();
+        }
+    }
+}
+
+fn print_version() {
+    println!("cargo2nix-{}\n", env!("CARGO_PKG_VERSION"));
+}
+fn print_help() {
+    print_version();
+    println!("$ cargo2nix                        # Print the help");
+    println!("$ cargo2nix -s,--stdout            # Output to stdout");
+    println!("$ cargo2nix -f,--file              # Output to Cargo.nix");
+    println!("$ cargo2nix -f,--file <file>       # Output to the given file");
+    println!("$ cargo2nix -v,--version           # Print version of cargo2nix");
+    println!("$ cargo2nix -h,--help              # Print the help");
+}
+
+fn write_to_file(file: impl AsRef<Path>) {
+    if file.as_ref().exists() {
+        print!(
+            "warning: do you want to overwrite '{}'? yes/no: ",
+            file.as_ref().display()
+        );
+        io::Write::flush(&mut io::stdout().lock()).expect("flush stdout buffer");
```

Comment:
This lock is unnecessary.
```suggestion
        io::Write::flush(&mut io::stdout()).expect("flush stdout buffer");
```

Reply:
Done